src/IonReader.ts

```diff
@@ -41,6 +41,17 @@ export type ReaderScalarValue =
  * invoke [[next]] to position a newly created reader on the first value.
  */
 export interface Reader {
+  /**
+   * Return the position of the current reader.
+   * The position refers to the character distance between the char where the reader
+   * stared (e.g. the first character of the file), and the current character the reader
+   * is reading.
+   *
+   * @returns a [[number]] type presenting the position of the character the reading is
+   * currently reading.
+   */
+  position(): number | null;
```

Comment:
A few thoughts here:
* What does it mean for a Reader's `position` to be `null`?
* The `number` returned by `position()` needs to make sense for both text and binary Readers. The comment describes only the meaning for a text reader.
* Unfortunately, the definition of a "character" [is ambiguous in Unicode terms](https://stackoverflow.com/a/27331885/109549). It could refer to a code point, code unit, glyph or grapheme cluster (among other possibilities). I think our best option here is to refer to code units:
  
  > A **code unit** is the unit of storage of a part of an encoded code point. In UTF-8 this means 8-bits, in UTF-16 this means 16-bits. A single code unit may represent a full code point, or part of a code point. For example, the snowman glyph (☃) is a single code point but 3 UTF-8 code units, and 1 UTF-16 code unit.

Reply:
Thanks for the input! 
- for default value, I took a second look, and actually once the `TextReader`/`BinaryReader` is initialized, the `position` will be initialized as 0 in `StringSpan`/`BinarySpan`. So I guess the `null` actually doesn't make sense.
- Comment updated, how about the below:
```suggestion
  /**
   * Return the position of the current reader.
   * The position refers to the distance between the code units where the reader
   * stared (e.g. the first code unit of the file), and the current code unit the 
   * reader is reading.
   * 
   * A code unit is the unit of storage of a part of an encoded code point.
   * Ref: https://stackoverflow.com/a/27331885/109549
   *
   * @returns a [[number]] type presenting the position of the code unit the reader is
   * currently reading.
   */
  position(): number;
```

Reply:
The doc comment is an improvement, but we still need a couple of tweaks.

* Because the `Reader` interface is used for both text and binary, we need to explain what `position()` means when the `Reader` is a binary reader. Namely, that it will be the byte offset from the start of the input rather than the code unit offset.
* Because code units are different in different unicode encodings (UTF8 code units are 1-4 bytes, UTF16 are 2 bytes), we should specify that this method returns a number UTF-16 code units, the encoding used by JavaScript strings.
* Even though it's a good explanation, I'd like to avoid using a StackOverflow link in the public documentation. We can point to [JavaScript's String length documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/length#Description), however. We could also refer users to [chapter 2.5 of the Unicode Standard, "Encoding Forms"](https://www.unicode.org/versions/Unicode13.0.0/ch02.pdf), but that's a bit dense for casual reference.
* We should warn users that `Reader`s cannot safely skip to a given position in the stream and begin reading as they may be skipping over system values like symbol definitions.


What do you think of:
```typescript
/**
   * Returns the Reader's offset from the beginning of its input.
   * 
   * For binary Readers, the return value is the number of bytes that have
   * been processed.
   * 
   * For text Readers, the return value is the number of UTF-16 code units
   * that have been processed, regardless of the input's original encoding.
   * For more on JavaScript's in-memory representation of text, see:
   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/length#Description
   * 
   * Note that a Reader cannot safely skip to a given position in input without
   * processing the stream leading up to that position. This is because there are
   * mid-stream system level values that must be processed to guarantee that the 
   * Reader is in a valid state. It is safe, however, to start at the beginning of a data
   * source and call next() until you reach the desired position, as the reader
   * will still have the opportunity to process system-level values along the way.)
   *
   * @returns the [[number]] of bytes or UTF-16 code units that the reader has processed.
  position(): number;
```
?

Reply:
Thanks for the detailed reply! This looks a lot better! 
I didn't think that much when I started out, and referring to StackOverflow was really a miss 🤦🏻‍♂️
I will update the commit with the latest comment.